verilog/tools/ls/autoexpand.cc

```diff
@@ -709,9 +724,17 @@ void AutoExpander::Module::SortPortsByLocation() {
 void AutoExpander::Module::RetrieveAutoTemplates() {
   absl::string_view autotmpl_search_span = StringSpanOfSymbol(symbol_);
   absl::string_view autotmpl_span;
+  absl::string_view autotmpl_inst_name;
   while (RE2::FindAndConsume(&autotmpl_search_span, *autotemplate_re_,
-                             &autotmpl_span)) {
+                             &autotmpl_span, &autotmpl_inst_name)) {
     Template tmpl{.it = autotmpl_span.begin()};
+    if (!autotmpl_inst_name.empty()) {
+      tmpl.instance_name_re = std::make_shared<RE2>(autotmpl_inst_name);
+      if (!tmpl.instance_name_re->ok()) {
+        LOG(ERROR) << "Invalid regex in AUTO template: " << autotmpl_inst_name;
```

Comment:
Given that this is user-supplied data: Do you have a test that explicitly passes a broken regular expression to see if  everything else around handles this situation gracefully ?

Reply:
Good point, I'll add a test

Reply:
Done